src/Compilers/CSharp/Portable/Symbols/TupleTypeSymbol.cs

```diff
@@ -73,31 +82,154 @@ internal static TupleTypeSymbol ConstructTupleTypeSymbol(TupleTypeSymbol origina
+
+                    var field = new TupleFieldSymbol(elementNames.IsDefault ? TupleMemberName(elementIndex) : elementNames[elementIndex],
+                                               this,
+                                               elementTypes[elementIndex],
+                                               OneOrMany.Create(underlyingField?.AsMember(underlyingTypeChain[0])));
+                    fieldsBuilder.Add(field);
+                }
+            }
+            else
+            {
+                var maxNumLinks = NumberOfValueTuples(numElements);
+
+                // long tuple: each member gets a chain of field symbols
+                for (int elementIndex = 0; elementIndex < numElements; elementIndex++)
+                {
+                    var numLinks = NumberOfValueTuples(elementIndex + 1);
+                    var chainBuilder = ArrayBuilder<FieldSymbol>.GetInstance(numLinks);
+
+                    for (var linkIndex = 0; linkIndex < numLinks; linkIndex++)
+                    {
+                        var linkArity = (linkIndex != maxNumLinks - 1) ? 8 : SizeOfLastValueTuples(numElements);
+                        var linkPosition = (linkIndex != numLinks - 1) ? 7 : elementIndex % 7;
+
+                        WellKnownMember linkType = GetTupleTypeMember(linkArity, linkPosition);
+                        var linkField = (FieldSymbol)Binder.GetWellKnownTypeMember(binder.Compilation, linkType, diagnostics, syntax: syntax);
+                        var link = linkField?.AsMember(underlyingTypeChain[linkIndex]);
+
+                        chainBuilder.Add(link);
+                    }
+
+                    var field = new TupleFieldSymbol(elementNames.IsDefault ? TupleMemberName(elementIndex) : elementNames[elementIndex],
+                                               this,
+                                               elementTypes[elementIndex],
+                                               OneOrMany.Create(chainBuilder.ToImmutableAndFree()));
+                    fieldsBuilder.Add(field);
+                }
+            }
 
-            var tupleType = binder.GetWellKnownType(GetTupleType(numElements), diagnostics, syntax);
-            NamedTypeSymbol underlyingType = tupleType.Construct(elementTypes);
+            fields = fieldsBuilder.ToImmutableAndFree();
+            return underlyingTypeChain;
+        }
 
-            var fieldSymbols = ArrayBuilder<TupleFieldSymbol>.GetInstance(numElements);
+        /// <summary>
+        /// Get this tuple's constructor(s).
+        ///
+        /// For short tuples, a single constructor is returned.
+        /// For long tuples, a chain of constructors for nested types (largest first).
+        /// </summary>
+        internal ImmutableArray<MethodSymbol> GetTupleConstructorChain(TupleExpressionSyntax node, Binder binder, DiagnosticBag diagnostics, ref bool hasErrors)
```

Comment:
I moved the constructor logic into `RewriteTupleCreationExpression` (part of `LocalRewriter`). If a constructor is missing it is able to report diagnostics properly. Let me know if that's ok.